internal/manifests/mutate.go

```diff
@@ -170,8 +170,9 @@ func mutateRoleBinding(existing, desired *rbacv1.RoleBinding) {
 }
 
 func mutateServiceMonitor(existing, desired *monitoringv1.ServiceMonitor) {
-	// ServiceMonitor selector is immutable so we set this value only if
```

Comment:
I tried to mutate all fields of ServiceMonitor instances manually (kubectl edit ...) and it worked fine, so I'm not sure why this comment is here. I also couldn't find anything substantial when searching for "immutable" in the prometheus-operator repository on GitHub.